### ISSUE TITLE
Issues Syncing Blobs which contain whitespace in the name

### ISSUE THREAD

**Issue by TylrRssll** [Azure/azure-storage-azcopy#2384]
### Which version of the AzCopy was used? 
10.20.1

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows (Azure VM)

### What command did you run?
`
az login
`
Logged in as my own AD account which has the RBAC roles "Storage Blob Data Owner", "Storage Queue Data Owner" on both source and destination storage accounts.

Then 
`
azcopy copy 'https://devstorageaccount.blob.core.windows.net/gram-stain/rt-test' 'https://prodstorageaccount.blob.core.windows.net/gram-stain/' --recursive
`

Note the storage account names have been changed to obscure the real names

### What problem was encountered?
`
INFO: Transfers could fail because AzCopy could not verify if the destination supports tiers.
INFO: Authentication failed, it is either not correct, or expired, or does not have the correct permission -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, C:/Users/VssAdministrator/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.15.0/azblob/zc_storage_error.go:42
===== RESPONSE ERROR (ServiceCode=CannotVerifyCopySource) =====
Description=This request is not authorized to perform this operation.
RequestId:da772175-d01e-0040-09ab-ef6190000000
Time:2023-09-25T12:27:27.9712854Z, Details:
   Code: CannotVerifyCopySource
   PUT https://prodstorageaccount.blob.core.windows.net/gram-stain/rt-test/test/This%20file%20has%20spaces.txt?timeout=901
   Authorization: REDACTED
   Content-Length: [0]
   User-Agent: [AzCopy/10.20.1 Azure-Storage/0.15 (go1.19.12; Windows_NT)]
   X-Ms-Access-Tier: [Hot]
   X-Ms-Blob-Cache-Control: []
   X-Ms-Blob-Content-Disposition: []
   X-Ms-Blob-Content-Encoding: []
   X-Ms-Blob-Content-Language: []
   X-Ms-Blob-Content-Type: [text/plain]
   X-Ms-Blob-Type: [BlockBlob]
   X-Ms-Client-Request-Id: [c24d7ce5-7c23-442e-50c4-574695aa43d5]
   X-Ms-Copy-Source: [https://devstorageaccount.blob.core.windows.net/gram-stain/rt-test/test/This%20file%20has%20spaces.txt]
   X-Ms-Copy-Source-Authorization: REDACTED
   X-Ms-Version: [2020-10-02]
   --------------------------------------------------------------------------------
   RESPONSE Status: 403 This request is not authorized to perform this operation.
   Content-Length: [248]
   Content-Type: [application/xml]
   Date: [Mon, 25 Sep 2023 12:27:27 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Client-Request-Id: [c24d7ce5-7c23-442e-50c4-574695aa43d5]
   X-Ms-Error-Code: [CannotVerifyCopySource]
   X-Ms-Request-Id: [da772175-d01e-0040-09ab-ef6190000000]
   X-Ms-Version: [2020-10-02]
`
### How can we reproduce the problem in the simplest way?
1. Create two storage accounts (source and destination)
2. Create the same container on both storage accounts
3. Grant "Storage Blob Data Owner" on both source and dest accounts to the identity used for syncing
4. "Grant Storage Queue Data Owner" on both source and dest accounts to the identity used for syncing
5. Create a folder to the source container 
6. Add a file with no spaces in the name and sync the containers using the az copy command, which should work
7. Add another file with spaces in the name and repeat the az copy command, which should fail

### Have you found a mitigation/solution?
Removing or renaming the files \ folders which have spaces in the name allos the copy to complete,  this is not practical for our use case though as the source container has thousands of files with spaces in the name we need to sync to the destination container.

**Comment by nakulkar-msft** [comment]
Hi @TylrRssll, do you have firewall enabled on source?

**Comment by TylrRssll** [comment]
Hi, the storage account is not publicly accessible via the internet.

Both accounts have been set to allow access via private endpoints or from specific VNET subnets, the VM which is being used to connect to the storage accounts has been added to the list of permitted subnets, and the subnet has a service connection defined to Microsoft.Storage.

If it was a firewall issue, I would have expected this to be an issue for all folders and files within the source storage account, rather than only those which a space character in the folder \ file name.

**Comment by nakulkar-msft** [comment]
Hi @TylrRssll, can you forward the log file to me. My email is on my profile. Thanks.

**Comment by TylrRssll** [comment]
I will get this to you ASAP,  possibly today, if not tomorrow

**Comment by TylrRssll** [comment]
Mail sent with the logs.

**Comment by dcorrea-wisr** [comment]
have you found a solution to this? we need to copy over 30 thousand blobs, it is not practical to rename them

**Comment by gapra-msft** [comment]
@TylrRssll were you able to get resolution to your issue via email?

**Comment by TylrRssll** [comment]
@dcorrea-wisr & @gapra-msft 
I don't have the email any more to reference,  but what we did in the end was a workaround.  From memory this was it.

1. Created a subnet within our VNET with service endpoints enabled to azure storage 
2. Created a VM in the subnet with a 1TB secondary storage disk attached.
3. On the VM we used blobfuseV2 to mount each of the storage containers as mount points on the VM:  https://learn.microsoft.com/en-us/azure/storage/blobs/blobfuse2-how-to-deploy?tabs=RHEL
4. We then copied the data across the mounted file systems which sync'd the data from from the source to target
5. Once complete we destroyed the VM and additional disks.
